src/hotspot/share/opto/library_call.cpp

```diff
@@ -6962,6 +6964,39 @@ bool LibraryCallKit::inline_base64_decodeBlock() {
   return true;
 }
 
+bool LibraryCallKit::inline_poly1305_processBlocks() {
+  address stubAddr;
+  const char *stubName;
+  assert(UsePolyIntrinsics, "need Poly intrinsics support");
+  assert(callee()->signature()->size() == 5, "poly1305_processBlocks has %d parameters", callee()->signature()->size());
+  stubAddr = StubRoutines::poly1305_processBlocks();
+  stubName = "poly1305_processBlocks";
+
+  if (!stubAddr) return false;
+  Node* input = argument(1);
```

Comment:
Receiver null check is missing. Since the method being intrinsified is non-static, the intrinsic itself has to take care of receiver null check.

Reply:
I think I found the right code to copy-paste, if you could check again pls.